### ISSUE TITLE
'THREAD_INFORMATION_CLASS' has not been declared

### ISSUE THREAD

**Issue by pqqsxhy** [arvidn/libtorrent#7321]
**Please provide the following information**

libtorrent version (or branch):
2.08
platform/architecture:
windows
compiler and compiler version:
mingy.  gcc
please describe what symptom you see, what you would expect to see instead and
how to reproduce it.

cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD=14 -Dstatic_runtime=ON -NDEBUG -DBoost_INCLUDE_DIR=C:\msys64\mingw64\include -G Ninja ..

ninja -j8

**Comment by arvidn** [comment]
I don't believe Mingw builds are covered by CI right now. I'll try to add coverage, without it it will likely just keep breaking in the future.

**Comment by arvidn** [comment]
this type is supposed to be available when including `Windows.h, see [msdn](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/ne-processthreadsapi-thread_information_class).

Do you understand why it's not available on mingw?

**Comment by arvidn** [comment]
I'm having a hard time building with mingw in CI. Do you have any idea of how to address this?
https://github.com/arvidn/libtorrent/actions/runs/4280097493/jobs/7451473630#step:7:88

**Comment by luzpaz** [comment]
@pqqsxhy ping

**Comment by arvidn** [comment]
I have added a mingw *build* step in CI. apparently it still creates broken executables, I'm not sure why.

**Comment by pqqsxhy** [comment]
> this type is supposed to be available when including `Windows.h, see [msdn](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/ne-processthreadsapi-thread_information_class).
> 
> Do you understand why it's not available on mingw?

Thankyou very much. I compiled it. Because my mingw version is too low, the windows.h winbase.h file is inconsistent with what libtorrent needs. Later, I copied these two files from other higher versions of mingw, and the compilation was normal.
